app/controllers/uploads_controller.rb

```diff
@@ -0,0 +1,28 @@
+class UploadsController < ApplicationController
+  # GET /uploads/signature
+  def signature
+    date_stamp = Date.strptime(params[:datetime], "%Y%m%dT%H%M%SZ").strftime("%Y%m%d")
+
```

Comment:
We think that OpenSSL is encrypting these keys for AWS. True? False? @radical-ube

Reply:
Can of worms ... but this is implementing the [AWS v4 signature algorithm](https://docs.aws.amazon.com/AmazonS3/latest/API/sig-v4-authenticating-requests.html).

And because the javascript upload-file-to-S3 library we use ([EvaporateJS](https://github.com/TTLabs/EvaporateJS)) is real picky, I couldn't use the AWS SDK to do this.  I had to manually port some of their [example signing code](https://github.com/TTLabs/EvaporateJS/blob/master/example/signing_example_awsv4_controller.rb) - though not exactly.

Honestly, can't remember how/why Ryan landed on this.  But it works!